gusto/diagnostics.py

```diff
@@ -53,7 +55,10 @@ def l2(f):
 
     @staticmethod
     def total(f):
-        return assemble(f * dx)
+        if len(f.ufl_shape) == 0:
+            return assemble(f * dx)
+        else:
+            pass
```

Comment:
I'm a bit confused - would these classes be defined within the current `Diagnostics` class?